### PR TITLE
Nexus: Well know error translation

### PR DESCRIPTION
**Pull request by Quinn-With-Two-Ns**
Translate some well know errors like what we did with the Go SDK. This does not cover all possible SDK exception, just some we except users to throw in this initial release.

closes https://github.com/temporalio/sdk-java/issues/2209

### CODE REVIEW

File: temporal-sdk/src/test/java/io/temporal/workflow/nexus/OperationFailureConversionTest.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+  @Rule
+  public SDKTestWorkflowRule testWorkflowRule =
+      SDKTestWorkflowRule.newBuilder()
+          .setWorkflowTypes(TestNexus.class)
+          .setNexusServiceImplementation(new TestNexusServiceImpl())
+          .build();
+
+  @Before
+  public void checkRealServer() {
```
Comment (Quinn-With-Two-Ns) [inline temporal-sdk/src/test/java/io/temporal/workflow/nexus/OperationFailureConversionTest.java:54]: @pdoerner The test server does not appear to match the real server's behaviour here

File: temporal-sdk/src/main/java/io/temporal/client/WorkflowException.java [hunk 2]
```diff
@@ -59,7 +59,6 @@ private static String getMessage(WorkflowExecution execution, String workflowTyp
         + execution.getWorkflowId()
         + "', runId='"
         + execution.getRunId()
-        + (workflowType == null ? "" : "', workflowType='" + workflowType + '\'')
-        + '}';
+        + (workflowType == null ? "" : "', workflowType='" + workflowType + '\'');
```
Comment (Quinn-With-Two-Ns) [inline temporal-sdk/src/main/java/io/temporal/client/WorkflowException.java:62]: Not sure what this random `}` was for

File: temporal-sdk/src/test/java/io/temporal/workflow/nexus/OperationFailureConversionTest.java (hunk 1 above)
Comment (Quinn-With-Two-Ns) [reply temporal-sdk/src/test/java/io/temporal/workflow/nexus/OperationFailureConversionTest.java]: Thanks PJ!

File: temporal-sdk/src/main/java/io/temporal/internal/nexus/NexusTaskHandlerImpl.java [hunk 3]
```diff
@@ -177,12 +179,34 @@ private CancelOperationResponse handleCancelledOperation(
...
+  private void convertKnownFailures(Throwable failure) {
+    if (failure instanceof ApplicationFailure) {
+      if (((ApplicationFailure) failure).isNonRetryable()) {
+        throw new OperationHandlerException(
+            OperationHandlerException.ErrorType.BAD_REQUEST, failure.getMessage());
+      }
+      throw new OperationHandlerException(
+          OperationHandlerException.ErrorType.INTERNAL, failure.getMessage());
+    }
+    if (failure instanceof WorkflowException) {
```
Comment (bergundy) [inline temporal-sdk/src/main/java/io/temporal/internal/nexus/NexusTaskHandlerImpl.java:200]: Can you confirm that this roughly maps to `WorkflowExecutionError` or `QueryRejectedError` in Go?

This is fine as an initial first pass at this but I think eventually we may want a different approach. We'll see if we get any feedback on the current approach.

**Review by bergundy — APPROVED** [review]

File: temporal-sdk/src/main/java/io/temporal/internal/nexus/NexusTaskHandlerImpl.java (hunk 3 above)
Comment (Quinn-With-Two-Ns) [reply temporal-sdk/src/main/java/io/temporal/internal/nexus/NexusTaskHandlerImpl.java:200]: `WorkflowQueryException` and `WorkflowFailedException` extend `WorkflowException`

Comment (Quinn-With-Two-Ns) [reply temporal-sdk/src/main/java/io/temporal/internal/nexus/NexusTaskHandlerImpl.java:200]: So they are both treated as `BAD_REQUEST`

**Pull request merged**